File: mahasiswa.cpp
//import library
#include <iostream>
#include <string>

using namespace std;

//deklarasi class
class Mahasiswa
{
    //atribut privat
    private:
        string nama;
        string nim;
        string prodi;
        string fakultas;

    //public methods
    public:

        //constructor
        //tidak memiliki tipe data return
        Mahasiswa()
        {
            //set value semua properti
            this->nama = "";
            this->nim = "";
            this->prodi = "";
            this->fakultas = "";
        }

        Mahasiswa(string nama, string nim, string prodi, string fakultas)
        {
            this->nama = nama;
            this->nim = nim;
            this->prodi = prodi;
            this->fakultas = fakultas;
        }

        /*getter dan setter*/

        //get nama
        string get_nama()
        {
            return this->nama;
        }

        //set nama
        void set_nama(string nama)
        {
            this->nama = nama;
        }

        //get nim
        string get_nim()
        {
            return this->nim;
        }

        //set nim
        void set_nim(string nim)
        {
            this->nim = nim;
        }

        //get prodi
        string get_prodi()
        {
            return this->prodi;
        }

        //set prodi
        void set_prodi(string prodi)
        {
            this->prodi = prodi;
        }

        //get fakultas
        string get_fakultas()
        {
            return this->fakultas;
        }

        //set fakultas
        void set_fakultas(string fakultas)
        {
            this->fakultas = fakultas;
        }

        /*public methods*/

        //insert
        void insert()
        {
            cout << this->nama << "is inserted" << '\n';
        }

        //edit
        void edit()
        {
            cout << this->nama << "is edited" << '\n';
        }

        //delete
        void insert()
        {
            cout << this->nama << "is edited" << '\n';
        }

        /* Destructors */
        //default destructors
        ~Mahasiswa()
        {

        }
};

File: mahasiswa.py
#nama, NIM, Program studi, fakultas
class Mahasiswa:
    __nama = ""
    __nim = ""
    __prodi = ""
    __fakultas = ""
    
    def __init__(self, nama = "", nim = "", prodi = "", fakultas = ""):
        self.__nama = nama
        self.__nim = nim
        self.__prodi = prodi
        self.__fakultas = fakultas
    
    #get nama
    def get_nama(self):
        return self.__nama
    
    #set nama
    def set_nama(self, nama):
        self.__nama = nama
        
    #get nim
    def get_nim(self):
        return self.__nim
    
    #set nim
    def set_nim(self, nim):
        self.__nim = nim
        
    #get prodi
    def get_prodi(self):
        return self.__prodi
    
    #set prodi
    def set_prodi(self, prodi):
        self.__prodi = print
        
    #get fakultas
    def get_fakultas(self):
        return self.__fakultas
    
    #set fakultas
    def set_fakultas(self, fakultas):
        self.__fakultas = fakultas
        
    #public methods
    
    #insert
    def insert(self):
        print(self.__nama)
        
    #edit
    def edit(self):
        print(self.__nama)
        
    #delete
    def delete(self):
        print(self.__nama)
        
    